Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-password-form/ContestPasswordForm.tsx

```diff
@@ -1,92 +1,65 @@
 
 import styles from './ContestPasswordForm.module.scss';
 
 interface IContestPasswordFormProps {
-    id: number;
-    isOfficial: boolean;
+  contestName: string;
+  onSuccess: () => void;
+  hasConfirmedParticipation: boolean;
 }
 
-const ContestPasswordForm = ({ id, isOfficial }: IContestPasswordFormProps) => {
-    const {
-        state: {
-            contest,
-            contestPasswordError,
-        },
-        actions: { submitPassword },
-    } = useCurrentContest();
-    const [ passwordValue, setPasswordValue ] = useState<string>('');
+const ContestPasswordForm = (props: IContestPasswordFormProps) => {
+    const { contestName, onSuccess, hasConfirmedParticipation } = props;
 
-    const { actions: { setPageTitle } } = usePageTitles();
+    const { themeColors, getColorClassName } = useTheme();
+    const { contestId: id, participationType } = useParams();
 
-    const passwordFieldName = 'contestPassword';
+    const [ password, setPassword ] = useState<string>('');
+    const [ errorMessage, setErrorMessage ] = useState<string>('');
+    const [ isLoading, setIsLoading ] = useState<boolean>(false);
 
-    const handleOnSubmitPassword = useCallback(async () => {
-        await submitPassword({ id, isOfficial, password: passwordValue });
-    }, [ id, isOfficial, passwordValue, submitPassword ]);
+    const [ registerUserForContest ] = useRegisterUserForContestMutation();
 
-    const handleOnChangeUpdatePassword = useCallback(
-        (value?: IFormControlOnChangeValueType) => {
-            setPasswordValue(isNil(value)
-                ? ''
-                : value.toString());
-        },
-        [ setPasswordValue ],
-    );
-
-    const renderErrorSpan = useCallback(
-        (message: string) => (
-            <span className={styles.errorMessage}>{message}</span>
-        ),
-        [],
-    );
-
-    const renderErrorMessage = useCallback(
-        () => {
-            if (!isNil(contestPasswordError)) {
-                const { detail } = contestPasswordError;
-                return renderErrorSpan(detail);
-            }
+    const textColorClassName = getColorClassName(themeColors.textColor);
+    const isOfficial = participationType === 'compete';
 
-            return null;
-        },
-        [ contestPasswordError, renderErrorSpan ],
-    );
-
-    useEffect(
-        () => {
-            if (!isNil(contest)) {
-                setPageTitle('Enter Contest Password');
-            }
-        },
-        [ contest, setPageTitle ],
-    );
+    const onPasswordSubmit = async () => {
+        setIsLoading(true);
+        const response = await registerUserForContest({ id: Number(id), isOfficial, password, hasConfirmedParticipation });
+        setPassword('');
+        if ((response as any).error) {
+            const { data } = (response as any).error;
+            setErrorMessage(data);
+        } else {
+            onSuccess();
+        }
+        setIsLoading(false);
```

Comment:
It would be good to clear the error message if we resubmit a new password, so it is more clear if the password is again wrong or that the password is submitted in general

Reply:
nice catch